DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/CourseSetupController.cs

```diff
@@ -13,21 +16,83 @@
     [Route("/TrackingSystem/CourseSetup")]
     public class CourseSetupController : Controller
     {
+        private const string CourseFilterCookieName = "CourseFilter";
+        private readonly ICourseCategoriesDataService courseCategoriesDataService;
         private readonly ICourseService courseService;
+        private readonly ICourseTopicsDataService courseTopicsDataService;
 
-        public CourseSetupController(ICourseService courseService)
+        public CourseSetupController(
+            ICourseService courseService,
+            ICourseCategoriesDataService courseCategoriesDataService,
+            ICourseTopicsDataService courseTopicsDataService
+        )
         {
             this.courseService = courseService;
+            this.courseCategoriesDataService = courseCategoriesDataService;
+            this.courseTopicsDataService = courseTopicsDataService;
         }
 
-        public IActionResult Index()
+        [Route("{page=1:int}")]
+        public IActionResult Index(
+            string? searchString = null,
+            string? sortBy = null,
+            string sortDirection = BaseSearchablePageViewModel.Ascending,
+            string? filterBy = null,
+            string? filterValue = null,
+            int page = 1
+        )
         {
+            if (filterBy == null && filterValue == null)
+            {
+                filterBy = Request.Cookies[CourseFilterCookieName];
+                if (filterBy == null)
+                {
+                    // Only show Active courses by default
+                    filterValue = CourseStatusFilterOptions.IsActive.FilterValue;
+                }

```

Comment:
Should this just be `filterBy = Request.Cookies[CourseFilterCookieName] ?? CourseStatusFilterOptions.IsActive.FilterValue`?